riscemu/core/int32.py

```diff
@@ -284,3 +308,6 @@ def shift_right_logical(self, amount: Union["Int32", int]) -> "UInt32":
         if isinstance(amount, Int32):
             amount = amount.value
         return UInt32(self.value >> amount)
+
+    def signed(self) -> "Int32":
+        return Int32(self)
```

Comment:
How should this handle overflow/negative values? It may be worth making this explicit.

Reply:
The `ctypes.c_[u]int32` this wraps handles them natively when constructing a `c_int32` from a `c_uint32`. I can make it more explicit by passing the ctypes type!